src/audio/pcm_converter/pcm_converter.c

```diff
@@ -35,10 +35,12 @@ int pcm_convert_as_linear(const struct audio_stream *source, uint32_t ioffset,
 		return -EINVAL;
 
 	while (i < samples) {
-		/* calculate chunk size */
-		N1 = audio_stream_bytes_without_wrap(source, r_ptr);
-		N2 = audio_stream_bytes_without_wrap(sink, w_ptr);
-		chunk = MIN(N1, N2) * s_size_out;
+		/* calculate chunk size, "">> ffs(x) - 1" is equal "/ x" here */
+		N1 = audio_stream_bytes_without_wrap(source, r_ptr) >>
+		     (ffs(s_size_in) - 1);
```

Comment:
Can you do the (ffs() -1) outside the loop since it looks like is constant.

Reply:
Updated in https://github.com/thesofproject/sof/pull/3231